main.py

```diff
@@ -45,118 +84,117 @@ def shutdown(self):
 
         If you have some cleanup procedure, insert it here.
         """
+        self.scheduler.shutdown()
 
     @rest('/v1', methods=['GET'])
+    def get_all_mw(self):
+        """Return all maintenance windows."""
+        maintenances = self.scheduler.list_maintenances()
+        return jsonify(converter.unstructure(maintenances)), 200
+
     @rest('/v1/<mw_id>', methods=['GET'])
-    def get_mw(self, mw_id=None):
-        """Return one or all maintenance windows."""
-        if mw_id is None:
-            return jsonify(
-                [maintenance.as_dict()
-                 for maintenance in self.maintenances.copy().values()]), 200
-        try:
-            return jsonify(self.maintenances[mw_id].as_dict()), 200
-        except KeyError:
-            raise NotFound(f'Maintenance with id {mw_id} not found')
+    def get_mw(self, mw_id: MaintenanceID):
+        """Return one maintenance window."""
+        window = self.scheduler.get_maintenance(mw_id)
+        if window:
+            return jsonify(converter.unstructure(window)), 200
+        raise NotFound(f'Maintenance with id {mw_id} not found')
 
     @rest('/v1', methods=['POST'])
     def create_mw(self):
         """Create a new maintenance window."""
-        now = datetime.datetime.now(pytz.utc)
         data = request.get_json()
         if not data:
-            raise UnsupportedMediaType('The request does not have a json.')
+            raise UnsupportedMediaType('The request does not have a json')
         try:
-            maintenance = MW.from_dict(data, self.controller)
-        except ValueError as err:
-            raise BadRequest(f'{err}')
-        if maintenance is None:
-            raise BadRequest('One or more items are invalid')
-        if maintenance.start < now:
-            raise BadRequest('Start in the past not allowed')
-        if maintenance.end <= maintenance.start:
-            raise BadRequest('End before start not allowed')
+            maintenance = converter.structure(data, MW)
```

Comment:
@Ktmi what's the expected behavior if a non active or non existing switch (interface or link) id is present in any of the lists? that might be outside of the scope of this PR, but it's something that we need to align either in the blueprint or a new issue. This can also lead to invalid data being inserted into the collection, which although is not desirable, it might be acceptable if we make clear that's not being validated in this iteration that will be delivered.

Reply:
So, Ive been thinking about this one, and I think that we should allow ids for things we can't confirm to exist to be used. My main reasoning for this is that when we create a maintenance window, we won't be 100% certain of the state of the network's topology. It could be that switches where removed or added, making some ids that were previously invalid into valid, or valid into invalid.

Reply:
Right @Ktmi, I believe this would make sense for this iteration. It's probably something to be refined in the blueprint, just so network operators who will use in production are also aware of this, and then in the future we can keep evolving and even try to integrate more with `topology` as new requirements are discussed and aligned. Thanks.

Reply:
@viniarck and @Ktmi, the way I see is: we could have a parameter so that the operator informs Kytos that he/she is requesting an inexistent item on purpose and then the validation should be ignored. Something like `force` or `skip_check_exists` , etc.

Reply:
I've mapped Issue #65 to continue the discussion about this. Let's simplify for now and then we refine the requirement for the next release.